src/metric-engine/src/metadata_region.rs

```diff
@@ -62,17 +66,37 @@ const COLUMN_PREFIX: &str = "__column_";
 /// itself.
 pub struct MetadataRegion {
     pub(crate) mito: MitoEngine,
+    /// The cache for contents(key-value pairs) of region metadata.
+    ///
+    /// The cache should be invalidated when any new values are put into the metadata region or any
+    /// values are deleted from the metadata region.
+    cache: Cache<RegionId, RegionMetadataCacheEntry>,
     /// Logical lock for operations that need to be serialized. Like update & read region columns.
     ///
     /// Region entry will be registered on creating and opening logical region, and deregistered on
     /// removing logical region.
     logical_region_lock: RwLock<HashMap<RegionId, Arc<RwLock<()>>>>,
 }
 
+#[derive(Clone)]
+struct RegionMetadataCacheEntry {
+    key_values: Arc<BTreeMap<String, String>>,
+    size: usize,
+}
+
+const MAX_CACHE_SIZE: u64 = ReadableSize::mb(128).as_bytes();
+
 impl MetadataRegion {
     pub fn new(mito: MitoEngine) -> Self {
+        let cache = Cache::builder()
+            .max_capacity(MAX_CACHE_SIZE)
+            .eviction_policy(EvictionPolicy::lru())
+            .time_to_live(Duration::from_secs(60))
```

Comment:
Since we invalidate the cache with every update, we can extend its duration, maybe 5 mins or longer